torch/lib/libshm/socket.h

```diff
@@ -112,7 +113,7 @@ class ManagerServerSocket: public Socket {
   }
 
   virtual ~ManagerServerSocket() {
-    unlink(socket_path.c_str());
+    SYSCHECK_ERR_RETURN_NEG1(unlink(socket_path.c_str()));
```

Comment:
Before this PR, we don't have a correct file created, so the actual file is never unlinked.
By adding this check, we can have a better debugging result if anything invalid is introduced in the future.